crates/object-builder/src/buffer_stream.rs

```diff
@@ -0,0 +1,93 @@
+use anyhow::Result;
+use cdl_dto::edges::TreeResponse;
+use futures::Stream;
+use serde_json::Value;
+use std::{collections::HashMap, num::NonZeroU8, pin::Pin, task::Poll};
+
+mod buffer;
+
+use crate::{ObjectIdPair, RowSource};
+use buffer::ObjectBuffer;
+
+pub struct ObjectBufferedStream<S> {
+    buffer: ObjectBuffer,
+    input: S,
+}
```

Comment:
```suggestion
pub struct ObjectBufferedStream<S>
where
    S: Stream<Item = Result<(ObjectIdPair, Value)>> + Unpin,
{
    buffer: ObjectBuffer,
    input: S,
}
```
Creation of this object where S is not an Stream doesn't make much logic sense.

Reply:
While yes, can you create this object without using `::new()` constructor?